crates/turborepo-lib/src/commands/link.rs

```diff
@@ -0,0 +1,206 @@
+                    _ => {}
+                }
+            }
+        }
+        CachingStatus::OverLimit => return Err(anyhow!("usage limit")),
+        CachingStatus::Paused => return Err(anyhow!("spending paused")),
+        CachingStatus::Enabled => {}
+    }
+
+    fs::create_dir_all(base.repo_root.join(".turbo"))
+        .context("could not create .turbo directory")?;
+    base.repo_config_mut()?
+        .set_team_id(Some(team_id.to_string()))?;
+
+    let chosen_team_name = match selected_team {
+        SelectedTeam::User => user_display_name,
+        SelectedTeam::Team(team) => team.name.as_str(),
+    };
+
+    if modify_gitignore {
+        add_turbo_to_gitignore(&base)?;
+    }
+
+    println!(
+        "
+{}  Turborepo CLI authorized for {}
+
+{}
+    ",
+        base.ui.rainbow(">>> Success!"),
+        chosen_team_name,
+        GREY.apply_to("To disable Remote Caching, run `npx turbo unlink`")
+    );
+    Ok(())
+}
+
+fn should_enable_caching() -> Result<bool> {
+    let theme = ColorfulTheme::default();
+    Ok(Confirm::with_theme(&theme)
+        .with_prompt(
+            "Remote Caching was previously disabled for this team. Would you like to enable it \
+             now?",
+        )
+        .default(true)
+        .interact()?)
+}
+
+fn select_team<'a>(teams: &'a [Team], user_display_name: &'a str) -> Result<SelectedTeam<'a>> {
+    let mut team_names = vec![user_display_name];
+    team_names.extend(teams.iter().map(|team| team.name.as_str()));
+
+    let theme = ColorfulTheme {
+        active_item_style: Style::new().cyan().bold(),
+        active_item_prefix: Style::new().cyan().bold().apply_to(">".to_string()),
+        ..ColorfulTheme::default()
+    };
+    let selection = Select::with_theme(&theme)
+        .items(&team_names)
+        .default(0)
+        .interact()?;
+
+    if selection == 0 {
+        Ok(SelectedTeam::User)
+    } else {
+        Ok(SelectedTeam::Team(&teams[selection - 1]))
+    }
+}
+
+fn should_link(base: &CommandBase, location: &str) -> Result<bool> {
+    let prompt = format!(
+        "{}{} {}",
+        BOLD.apply_to(GREY.apply_to("? ")),
+        BOLD.apply_to("Would you like to enable Remote Caching for"),
+        base.ui.apply(BOLD.apply_to(CYAN.apply_to(location)))
+    );
+
+    Ok(Confirm::new().with_prompt(prompt).interact()?)
+}
+
+fn enable_caching(url: &str) -> Result<()> {
+    webbrowser::open(url).with_context(|| {
+        format!(
+            "Failed to open browser. Please visit {} to enable Remote Caching",
+            url
+        )
+    })?;
+
+    println!("Visit {} in your browser to enable Remote Caching", url);
+
+    // We return an error no matter what
+    return Err(anyhow!("link after enabling caching"));
+}
+
+fn add_turbo_to_gitignore(base: &CommandBase) -> Result<()> {
+    let gitignore_path = base.repo_root.join(".gitignore");
+
+    if !gitignore_path.exists() {
+        let mut gitignore = File::create(gitignore_path)?;
+        #[cfg(unix)]
```

Comment:
Should we include setting the permissions on macos as well?

Reply:
Isn't macOS also unix?